### PR TITLE
TT-506/2 - Adding info about self-healing

### PR DESCRIPTION
**Pull request by mativm02**
## Description

Adding information about the changes made in https://github.com/TykTechnologies/tyk-pump/pull/486
## Related Issue

https://tyktech.atlassian.net/browse/TT-506?atlOrigin=eyJpIjoiN2EwMWU5MGZlODVmNGUxN2I4ZTNmNzJmNWNlMWNhNTAiLCJwIjoiaiJ9
## Motivation and Context

We need to describe what self-healing is and how it works. The documentation is being added in an external PR to not delay the PR that contains the code.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: README.md
Comment (andyo-tyk) [inline README.md:349]: I think this should be 16MB, not 16Mb.
Is the error something specific to Mongo Aggregate Pump, or just a standard thing with MongoDB? (I'm guessing it's the latter)

I think this line could definitely be re-written to be clearer - and can propose different words if you can confirm the above.

For example, to something like this:

By default, the maximum size of a document in MongoDB is 16MB. If we try to update a document that has grown to this size, an error is received.

File: README.md
Comment (andyo-tyk) [inline README.md:350]: The Mongo Aggregate pump creates a new document in the database for each "aggregation period"; the length of that period is defined by `current_aggregation_time`. If, during that period (in minutes) the document grows beyond 16MB, the error will be received and no more records will be recorded until the end of the aggregation period (when a new document will be created).

The Self Healing option in the Mongo Aggregate Pump avoids this data loss by monitoring the size of the current document. When this hits the 16MB limit, Pump will automatically create a new document with the current timestamp and start writing to that instead.

When it does this, the Pump will halve the `current_aggregation_time ` so that it aggregates records for half the time period originally configured before creating a new document, reducing the risk of repeatedly hitting the maximum document size.

This self healing is repeatable, however, such that if the document size does reach maximum (16MB) even with the new shorter aggregation period, a new document will be created and the `current_aggregation_time` will be halved again.

The minimum value for `current_aggregation_time` is 1; Self Healing cannot reduce it beyond this value.

File: README.md
Comment (andyo-tyk) [inline README.md:351]: For example, if the `current_aggregation_time` is configured as 50 (minutes) but the document hits the maximum size (16MB), a new document will be started and the `current_aggregation_time` will be set to 25 (minutes).

File: README.md
Comment (andyo-tyk) [inline README.md:353]: Note that `store_analytics_per_minute` takes precedence over `current_aggregation_time` so if `store_analytics_per_minute` is equal to true, the value of `current_aggregation_time` will be equal to 1 and self healing will not operate.

File: README.md
Comment (mativm02) [reply README.md:349]: @andyo-tyk you're right, the error is just a standard limit set by MongoDB.

That new text looks good to me.

**Comment by tbuchaillot** [comment]
@andyo-tyk @marksou can you check this?

**Review by andyo-tyk — APPROVED** [review]
Sorry for the delay, I missed any reminder (if there was one) from github. Looks great - thank you.

**Comment by tbuchaillot** [comment]
@mativm02 what's up with this PR?

**Comment by mativm02** [comment]
> @mativm02 what's up with this PR?

I'm waiting for approval. 
I added you as a reviewer, could you take a look at it, please?

**Review by tbuchaillot — APPROVED** [review]

**Pull request merged**
